runtime/sema/elaboration.go

```diff
@@ -145,6 +146,7 @@ func NewElaboration(gauge common.MemoryGauge) *Elaboration {
 		ReferenceExpressionBorrowTypes:      map[*ast.ReferenceExpression]Type{},
 		IndexExpressionIndexedTypes:         map[*ast.IndexExpression]ValueIndexableType{},
 		IndexExpressionIndexingTypes:        map[*ast.IndexExpression]Type{},
+		ForceExpressionTypes:                map[*ast.ForceExpression]Type{},
```

Comment:
Maybe also gate this behind `if checker.lintingEnabled {`, so we don't have to unnecessarily allocate it